yarn-project/pxe/src/simulator_oracle/simulator_oracle.test.ts

```diff
@@ -37,6 +37,7 @@ import { type PxeDatabase } from '../database/index.js';
 import { KVPxeDatabase } from '../database/kv_pxe_database.js';
 import { ContractDataOracle } from '../index.js';
 import { SimulatorOracle } from './index.js';
+import { WINDOW_HALF_SIZE } from './tagging_utils.js';
```

Comment:
The changes in this file are cluttered by renaming of `senderOffset` to `tagIndex`. The only relevant changes in this file are on lines 295 until 312. There it can be seen that the optimization really took effect (getting half the requests on line 296.